src/verify_cert.rs

```diff
@@ -204,14 +204,34 @@ fn check_signatures(
 }
 
 fn check_crl(
-    _supported_sig_algs: &[&SignatureAlgorithm],
+    supported_sig_algs: &[&SignatureAlgorithm],
     cert: &Cert,
-    _issuer_subject: untrusted::Input,
-    _issuer_spki: untrusted::Input,
+    issuer_subject: untrusted::Input,
+    issuer_spki: untrusted::Input,
     crl_provider: &dyn CrlProvider,
 ) -> Result<(), Error> {
-    let _relevant_crl = crl_provider.crl_for_cert(cert);
-    todo!();
+    assert_eq!(cert.issuer, issuer_subject);
+
+    let relevant_crl = crl_provider.crl_for_cert(cert);
+    match relevant_crl {
+        None => Ok(()),
```

Comment:
Are we content with having a no-error interface here? I guess that might be sort of fundamental to CRLs, that not having a CRL gives the same result as getting the absence from a CRL? Do we think some callers might want to assert that a CRL was present for a trust anchor?

As an alternative, we could yield an `Option<CertificateUnrevoked>` token type, similar to the assertion types in rustls verification.

Style suggestions: I would inline the `crl_for_cert()` call and return the `None` case directly, as in

```rust
let crl = match crl_provider.crl_for_cert(cert) {
    Some(crl) => crl,
    None => return Ok(()),
};
```

Reply:
> Are we content with having a no-error interface here? I guess that might be sort of fundamental to CRLs, that not having a CRL gives the same result as getting the absence from a CRL?

I guess I'm not sure what the webpki code will do with that distinction. If there's no CRL, there's no revocation status. If there's a CRL with no entry for the cert in question, there's no revocation status. In the end the flow of execution will continue the same way :thinking: 

> As an alternative, we could yield an Option<CertificateUnrevoked> token type, similar to the assertion types in rustls verification.

I'm open to this. Even though I don't think it will change how the webpki side operates I can see an argument that it could avoid bugs on the side of the trait implementer and lets us use the type system to maximize clarity of intent.

> Style suggestions: I would inline the crl_for_cert() call and return the None case directly, as in

Sounds good :+1:

Reply:
> I guess I'm not sure what the webpki code will do with that distinction. If there's no CRL, there's no revocation status. If there's a CRL with no entry for the cert in question, there's no revocation status. In the end the flow of execution will continue the same way 🤔

Yeah... I'm not completely sure myself. Just that in terms of control flow, it could help clarify what's going on -- it can be easy to just return `Ok(())` without paying much attention to what it means.

Reply:
> Style suggestions: I would inline the crl_for_cert() call and return the None case directly, as in

Done.

> As an alternative, we could yield an Option token type, similar to the assertion types in rustls verification.

Done. `check_crl` returns ` Result<Option<CertNotRevoked>, Error> ` now. We return `Ok(None)` when no CRL was found to use for revocation status, `Ok(Some(CertNotRevoked::assertion()))` when there was a CRL checked and the cert was found to be not revoked, and `Err(Error::CertRevoked)` when there was a CRL checked and the cert was found to be revoked.